remote/remote_state_s3.go

```diff
@@ -78,29 +85,31 @@ func (s3Initializer S3Initializer) NeedsInitialization(config map[string]interfa
 // Initialize the remote state S3 bucket specified in the given config. This function will validate the config
 // parameters, create the S3 bucket if it doesn't already exist, and check that versioning is enabled.
 func (s3Initializer S3Initializer) Initialize(config map[string]interface{}, terragruntOptions *options.TerragruntOptions) error {
-	s3Config, err := parseS3Config(config)
+	s3ConfigExtended, err := parseExtendedS3Config(config)
 	if err != nil {
 		return err
 	}
 
-	if err := validateS3Config(s3Config, terragruntOptions); err != nil {
+	if err := validateS3Config(s3ConfigExtended, terragruntOptions); err != nil {
 		return err
 	}
 
+	var s3Config = s3ConfigExtended.remoteStateConfigS3
```

Comment:
Why the intermediate variable? Can't you use `s3ConfigExtended.Region`, `s3ConfigExtended.Endpoint`, etc below?

Reply:
No, s3configExtended has these fields only:

* remoteStateConfigS3
* S3BucketTags
* DynamotableTags

If I didn't have the intermediate variable, I would have to use `s3ConfigExtended.remoteStateConfigS3.Region`, which I find less readable.

My goal was to separate the actual terraform backend config from the two extra keys that are just for terragrunt.